scalafmt-core/shared/src/main/scala/org/scalafmt/util/ValidationOps.scala

```diff
@@ -3,19 +3,25 @@ package org.scalafmt.util
 import scala.collection.mutable
 
 object ValidationOps {
-  def assertNonNegative(ns: sourcecode.Text[Int]*): Unit = {
+
+  def addIfNegative(
+      ns: sourcecode.Text[Int]*
+  )(implicit errors: mutable.Buffer[String]): Unit =
     ns.foreach { n =>
       if (n.value < 0)
-        throw new IllegalArgumentException(
-          s"${n.source} must be non-negative, was ${n.value}"
-        )
+        errors += s"${n.source} must be non-negative, was ${n.value}"
     }
-  }
 
   def addIf(
       what: sourcecode.Text[Boolean],
       why: => String = ""
   )(implicit errors: mutable.Buffer[String]): Unit =
-    if (what.value) errors += what.source + why
+    addIfDirect(what.value, what.source + why)
```

Comment:
Is it a hack around

> sourcecode.Text[T]: when you want to take a value of type T, but also want to get the "source text" of that particular value. Note that if you have multiple statements in a {} block, sourcecode.Text will only capture the source code for the last expression that gets returned. This implicit is slightly experimental; be sure to report any bugs you find!

?

Only last block capture is source of #2098 problem.

Reply:
I see direct usage at 386, sorry